### PR TITLE
fix deprecated links

### PR DESCRIPTION
**Pull request by Ivana-Markovic**

### CODE REVIEW

File: docs/ci-cd-environment/machine-types.md [hunk 1]
```diff
@@ -143,4 +143,4 @@ Apple machine types can be paired with the [MacOS Mojave image][macos-mojave].
 
 [agent]: https://docs.semaphoreci.com/reference/pipeline-yaml-reference/#agent
 [ubuntu1804]: https://docs.semaphoreci.com/ci-cd-environment/ubuntu-18.04-image/
-[macos-mojave]: https://docs.semaphoreci.com/ci-cd-environment/macos-mojave-xcode-11-image/
+[macos-mojave]: https://docs.semaphoreci.com/ci-cd-environment/macos-xcode-11-image/
```
Comment (AleksandarCole) [inline docs/ci-cd-environment/machine-types.md:146]: Link text still says "macos-mojave", maybe we can change it to "macos-xcode11"

File: docs/programming-languages/javascript-and-node-js.md [hunk 2]
```diff
@@ -173,7 +173,7 @@ for details on pre-installed browsers and testing tools on Semaphore.
 [tutorial]: https://docs.semaphoreci.com/examples/node-js-and-typescript-continuous-integration/
 [demo-project]: https://github.com/semaphoreci-demos/semaphore-demo-javascript
 [ubuntu-javascript]: https://docs.semaphoreci.com/ci-cd-environment/ubuntu-18.04-image/#javascript-via-node-js
-[macos-javascript]: https://docs.semaphoreci.com/ci-cd-environment/macos-mojave-xcode-11-image/#javascript-via-node-js
+[macos-javascript]: https://docs.semaphoreci.com/ci-cd-environment/macos-xcode-11-image/#javascript-via-node-js
```
Comment (AleksandarCole) [inline docs/programming-languages/javascript-and-node-js.md:176]: This link is a bit different, it should be, there is no dash between node and js on the page so it should be: https://docs.semaphoreci.com/ci-cd-environment/macos-xcode-11-image/#javascript-via-nodejs

File: docs/reference/pipeline-yaml-reference.md [hunk 3]
```diff
@@ -2087,10 +2087,10 @@ YAML parser, which is not a Semaphore 2.0 feature but the way YAML files work.
 - [Machine Types](https://docs.semaphoreci.com/ci-cd-environment/machine-types/)
 
 [ubuntu1804]: https://docs.semaphoreci.com/ci-cd-environment/ubuntu-18.04-image/
-[macos-mojave-xcode11]: https://docs.semaphoreci.com/ci-cd-environment/macos-mojave-xcode-11-image/
+[macos-mojave-xcode11]: https://docs.semaphoreci.com/ci-cd-environment/macos-xcode-11-image/
```
Comment (AleksandarCole) [inline docs/reference/pipeline-yaml-reference.md:2090]: Same here, we should remove "mojave" from link test and leave "macos-xcode11"

Comment (Ivana-Markovic) [reply docs/reference/pipeline-yaml-reference.md:2090]: Thanks, great catches (what I've done is find / replace 🤦‍♀️) I'll correct this.

**Comment by AleksandarCole** [comment]
After consulting with Ivana, closing this as we already fixed it.

**Pull request closed without merging**